modules/hostserv/offer.c

```diff
@@ -298,7 +298,7 @@ hs_cmd_take(struct sourceinfo *si, int parc, char *parv[])
 		if (l->group != NULL && !myuser_is_in_group(si->smu, l->group))
 			continue;
 
-		if (!irccasecmp(l->vhost, host))
+		if (!host || !irccasecmp(l->vhost, host))
```

Comment:
This will behave unexpectedly if there's one vhost offered, but it's restricted by group. I think a proper solution needs a bit more rearranging.

Reply:
I've created #807 with what I think is a more complete way to do this.